File: go/08_integrity_failures.go
// A08 - Integrity Failures. VULN: gob/json decode of untrusted types + plugin load.
package main

import (
	"encoding/gob"
	"bytes"
)

// VULN: A08 - unsafe deserialization
func DecodeSession(b []byte) interface{} {
	var v interface{}
	_ = gob.NewDecoder(bytes.NewReader(b)).Decode(&v)
	return v
}

File: java/08_integrity_failures.java
// A08 - Integrity Failures. VULN: Java deserialization without filter.
import java.io.*;

public class IntegrityFailures {
  // VULN: A08 - untrusted ObjectInputStream
  public static Object loadSession(byte[] data) throws Exception {
    ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data));
    return ois.readObject();
  }
}
